features/steps/permissions_component.go

```diff
@@ -103,78 +103,51 @@ func getPermissionsBundle() *permissions.Bundle {
 	}
 }
 
-// NewPermissionsComponent initializes mock server and inmemory mongodb used for running component tests.
-func NewPermissionsComponent(mongoFeature *componenttest.MongoFeature) (*PermissionsComponent, error) {
+// NewPermissionsComponent initializes mock server and in-memory mongodb used for running component tests.
+func NewPermissionsComponent(mongoURI string) (*PermissionsComponent, error) {
 	f := &PermissionsComponent{
 		HTTPServer:     &http.Server{},
 		errorChan:      make(chan error),
 		ServiceRunning: false,
 	}
 
 	var err error
-
 	f.Config, err = config.Get()
 	if err != nil {
 		return nil, err
 	}
 
-	getMongoURI := fmt.Sprintf("localhost:%d", mongoFeature.Server.Port())
-	databaseName := utils.RandomDatabase()
+	f.Config.MongoDB.ClusterEndpoint = mongoURI
+	f.Config.MongoDB.Database = utils.RandomDatabase()
+	f.Config.MongoDB.Username, f.Config.Password = createCredsInDB(&f.Config.MongoDB)
 
-	f.Config.MongoConfig.Database = databaseName
-	f.Config.MongoConfig.BindAddr = getMongoURI
-	f.Config.MongoConfig.Username, f.Config.MongoConfig.Password, err = createCredsInDB(getMongoURI, databaseName)
-	if err != nil {
+	if f.MongoClient, err = mongo.NewMongoStore(context.Background(), f.Config.MongoDB); err != nil {
 		return nil, err
 	}
 
-	mongodb := &mongo.Mongo{}
-
-	if err := mongodb.Init(f.Config.MongoConfig); err != nil {
-		return nil, err
-	}
-
-	f.MongoClient = mongodb
-
 	f.ApiFeature = componenttest.NewAPIFeature(f.InitialiseService)
 
-	f.Config, err = config.Get()
-	if err != nil {
-		return nil, err
-	}
-
 	fakePermissionsAPI := setupFakePermissionsAPI()
 	f.Config.AuthorisationConfig.JWTVerificationPublicKeys = rsaJWKS
 	f.Config.AuthorisationConfig.PermissionsAPIURL = fakePermissionsAPI.URL()
 
 	return f, nil
 }
 
-func createCredsInDB(getMongoURI string, databaseName string) (string, string, error) {
-	username := "admin"
-	password, _ := uuid.NewV4()
-	mongoConnectionConfig := &dpMongoDriver.MongoConnectionConfig{
-		TLSConnectionConfig: dpMongoDriver.TLSConnectionConfig{
-			IsSSL: false,
-		},
-		ConnectTimeoutInSeconds: 15,
-		QueryTimeoutInSeconds:   15,
-
-		Username:        "",
-		Password:        "",
-		ClusterEndpoint: getMongoURI,
-		Database:        databaseName,
-	}
-	mongoConnection, err := dpMongoDriver.Open(mongoConnectionConfig)
+func createCredsInDB(mongoConfig *dpMongoDriver.MongoDriverConfig) (string, string) {
+	mongoConfig.Username, mongoConfig.Password = "", ""
```

Comment:
I can't see the reason to have this line here?

Reply:
It was in the code base so I left it there after fixing it up to work with the new stuff - I agree I don't see the purpose either, but it does no harm, and I didn't want to remove it since the original developer had put it in, and I didn't have a good reason to remove it

Reply:
Done